### DISCUSSION TITLE
flood can't connect to qbittorrent with https/ssl

### DISCUSSION THREAD

**Discussion by lilws** [jesec/flood#231] [Q&A] [answered]
Hi,
I recently from rtorrent return to qbit cause the tracker I've recently joined not really like rtorrent in their tracker. I have no problem connect rtorrent with ssl, but with qbittorrent, I can't connect it with https.

Is there anything I can do to fix it?

**Comment by jesec** [comment, marked as answer]
Like any sane program, Node.js runtime (and Flood by extension) enforces SSL certificate validity by default. If you want to connect Flood with qBittorrent API via SSL, you have to deploy an acceptable certificate.

**Reply by jesec** [reply]
And what do you mean exactly? rTorrent exclusively uses SCGI. There is no SSL/TLS capability.

**Reply by lilws** [reply]
Thank you, I created a self-signed certificate with openssl. I did the same with rtorrent and it works.

**Reply by jesec** [reply]
But there is no point where rTorrent would need that.

**Reply by lilws** [reply]
Ok, I will need to find a way to do that. Just enable a feature in qbittorrent and everything messed up :)

**Reply by jesec** [reply]
Additionally, I would advise you to raise the issue with the operator of that tracker, if such limitation on rTorrent is caused by an allowedlist/blocklist. rTorrent has been around since 2005. It correctly implemented BitTorrent protocol and other accepted extensions. It is trusted as the most stable torrent client by the community.

**Reply by lilws** [reply]
Thanks,
They offer me to use the stable version of rtorrent (I think it is 0.9.7), it kinds really make me confuse because they don't accept rtorrent 0.9.8. It cause me bunches of errors when connect to their tracker. Their user use only deluge or qbittorrent. Very small user uses 0.9.7. And find a good 0.9.7 running through docker is not very comfortable with me. I need to add a lot of extra configs in rtorrent. I might then fall back to qbittorrent for a moment until find a better way.

**Reply by jesec** [reply]
0.9.8 does not have notable changes that would affect the consensus-layer. It only contains trivial bug fixes. See https://github.com/rakshasa/rtorrent/releases/tag/v0.9.8. 

If 0.9.7 is acceptable, there is no reason to block 0.9.8. Additionally, 0.9.8 has been released in 2019, and it has been thoroughly tested by the community in the past few years. It is definitely more common in the BitTorrent community to accept 0.9.8. 

Deluge and qBittorrent are known to crack under pressure. I think your tracker operator ought to review their policies if they allow "one release per month" versions but not battle-hardened veterans like rTorrent.

**Reply by lilws** [reply]
Well, I tried to contact them about this issue, but it seems they don't care much since 90% of their user is follow the rules and their banned rules is very strict. When look at dl/ul speed and peer catch in rtorrent from their tracker, it is very poor performance. One I jumpt to qbit, it just so much different.

for a single torrent with over 1000 peers, I can find only 9 of them is using rtorrent. mostly qbit, transmission and deluge or utorrent.

**Reply by jesec** [reply]
Well. If a torrent client is not allowed by the tracker, the tracker will not respond with the list of peers, and, as a result, of course there will be no speed at all as peer connections could not be established. You have to get them to lift the restrictions first.

All BitTorrent clients implement a standard and interoperable protocol. The performance of a torrent client will not be negatively affected by its popularity. rTorrent, being a Unix-only torrent client at the moment, is normally used by the more experienced users and the most dedicated members of the community. It is perfectly capable of interacting with other torrent clients like qBittorrent, Transmission, Deluge or uTorrent.

**Reply by lilws** [reply]
I love to use rtorrent + flood because they both are so perfect with each other. This tracker admin have already state that if using qbit or deluge it is must faster then use any of the torrent client out there. Last time I use rtorrent latest, I suffer difficult to catch a peer for dl/ul. Even after get connected, the speed only stay at 10 Mbps for a few minute then get disconnected and all fell below 500 Kbps for the rest of the dl/ul session. If I tried to stop/start the torrent, their tracker even sent a banned client message to notice me and it definitely take couple hours to reconnect to the tracker. Thus I can't race with their rule related to seed and ratio things. Last time I report problems, admin said something about ssd failure on their side. Later I tried again but still got the same problems. I may tried this again now.

**Reply by lilws** [reply]
Ok, I tried it for an hour and most of problem about tracker report message is gone. But catch a peer is still a thing to consider.
Thanks for assist me, I may try acme.sh or something else to get verified cert.
